drivers/at86rf2xx/at86rf2xx.c

```diff
@@ -35,23 +35,17 @@
 #include "debug.h"
 
 
-void at86rf2xx_setup(at86rf2xx_t *dev, spi_t spi, spi_speed_t spi_speed,
-                     gpio_t cs_pin, gpio_t int_pin, gpio_t sleep_pin,
-                     gpio_t reset_pin)
+void at86rf2xx_setup(at86rf2xx_t *dev, at86rf2xx_params_t *params)
 {
     netdev2_t *netdev = (netdev2_t *)dev;
 
     netdev->driver = &at86rf2xx_driver;
     /* initialize device descriptor */
-    dev->spi = spi;
-    dev->cs_pin = cs_pin;
-    dev->int_pin = int_pin;
-    dev->sleep_pin = sleep_pin;
-    dev->reset_pin = reset_pin;
+    dev->params = *params;
```

Comment:
If I compile with -O0 I get the same size for both version on samr21-xpro and still an improvement of 4 bytes with the memcpy version on iotlab.